src/app/jobs.py

```diff
@@ -0,0 +1,93 @@
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from global_ctx import app  # Import global app context
+from app.models import Feed, Post
+from app import db, logger, config
+from app.feeds import refresh_feed
+from app.routes import download_post
+
+
+def run_refresh_all_feeds():
+    """Main entry point for refreshing all feeds."""
+    logger.info("run_refresh_all_feeds call")
+
+    if app is None:
+        raise RuntimeError("Flask application context is None")
+
+    with app.app_context():
+        refresh_all_feeds()
+
+
+def process_post(post):
+    """Process a single post within the app context."""
+    with app.app_context():
+        try:
+            logger.info(f"Processing post: {post.title} (ID: {post.id})")
+            download_post(post.guid)
+            logger.info(f"Post {post.title} (ID: {post.id}) processed successfully.")
+        except Exception as e:
+            logger.error(f"Error processing post {post.id}: {e}")
+
+
+def refresh_all_feeds():
+    """Refresh all feeds and process newly released episodes."""
+    logger.info("Scheduled job started: Refreshing all podcast feeds.")
+
+    try:
+        # Refresh each feed
+        feeds = Feed.query.all()
+        logger.info(f"Found {len(feeds)} feeds to refresh.")
+
+        for feed in feeds:
+            logger.info(f"Refreshing feed: {feed.title} (ID: {feed.id})")
+            refresh_feed(feed)
+
+        db.session.commit()
+        logger.info("All feeds refreshed and database updated.")
+
+        # Identify new posts
+        new_posts = Post.query.filter(Post.processed_audio_path == None).all()
+        logger.info(f"Found {len(new_posts)} new posts to download and process.")
+
+        if not new_posts:
+            logger.info("No new posts to process.")
+            return
+
+        # Automatically whitelist new episodes if enabled
+        if config.automatically_whitelist_new_episodes:
+            for post in new_posts:
+                if not post.whitelisted:
+                    post.whitelisted = True
+
+            try:
+                db.session.commit()
+                logger.info("All new posts have been automatically whitelisted.")
+            except Exception as e:
+                logger.error(f"Failed to automatically whitelist new posts: {e}")
+                db.session.rollback()
+                return
+
+        # Filter posts to only process those that are whitelisted
+        posts_to_process = Post.query.filter(
+            Post.processed_audio_path == None,
+            Post.whitelisted == True
+        ).all()
```

Comment:
instead of performing this filtering here can we process the posts by calling a codepath which does the check? If this doesn't happen in `download_post` already can you refactor to expose a function shared by that code path `download_post_if_whitelisted` or something and use that?

Reply:
To be honest, I spent most of this time hating flask contexts and the horse it rode in on, so once I finally got it to work in jobs I wasn't overly married to the idea of using everything y'all already had, it was easier for me to just write a monolithic block of code that did what I wanted